connector/src/main/java/org/geysermc/connector/network/translators/java/world/JavaChunkDataTranslator.java

```diff
@@ -55,12 +57,15 @@ public void translate(ServerChunkDataPacket packet, GeyserSession session) {
             ChunkUtils.updateChunkPosition(session, session.getPlayerEntity().getPosition().toInt());
         }
 
-        if (packet.getColumn().getBiomeData() == null) //Non-full chunk
+        if (packet.getColumn().getBiomeData() == null && !session.getConnector().getConfig().isCacheChunks()) //Non-full chunk without chunk caching
```

Comment:
Maybe worth storing `session.getConnector().getConfig().isCacheChunks()` as a static variable so it doesn't need to be fetched everytime?